Apromore-Database/src/main/resources/db/migration/changeLog-Data.yaml

```diff
@@ -210,4 +210,34 @@ databaseChangeLog:
                   value: "1"
               - column:
                   name: is_valid
-                  value: "1"
\ No newline at end of file
+                  value: "1"
+
+  - changeSet:
+      id: 20211130111000
+      author: janeh
+      comment: "migrate user role to analyst"
+      changes:
+        - delete:
+            tableName: user_role
+            where: roleid = 2 AND
+              userid IN (
+              SELECT * FROM (
+              SELECT userid
+              FROM `user_role`
+              WHERE roleid=2 OR roleid=4
+              GROUP BY userid
+              HAVING COUNT(roleid) > 1
+              )tblTmp
+              );
```

Comment:
Suggest reducing to:
```
where: roleid = 2 AND userid IN (
    SELECT userid
    FROM `user_role`
    WHERE roleid IN (2, 4)
    GROUP BY userid
    HAVING COUNT(roleid) > 1
);
```

Reply:
Made slight change to the suggested query since SQL doesn't allow the target table (user_roles) to be referenced directly in the FROM clause:

`where: roleid = 2 AND userid IN (
    SELECT userid
    FROM (SELECT * FROM `user_role`) AS user_role_copy
    WHERE roleid IN (2, 4)
    GROUP BY userid
    HAVING COUNT(roleid) > 1
);`